model-datastructure/src/commonMain/kotlin/org/modelix/model/LinearHistory.kt

```diff
@@ -61,3 +68,73 @@ class LinearHistory(val baseVersionHash: String?) {
         return CLVersion(hash.getValue(store), store)
     }
 }
+
+class LinearHistory(val baseVersionHash: String?) {
+
+    val version2directDescendants: MutableMap<Long, Set<Long>> = HashMap()
+    val versions: MutableMap<Long, CLVersion> = HashMap()
+
+    /**
+     * Oldest version first
+     */
+    fun load(vararg fromVersions: CLVersion): List<CLVersion> {
```

Comment:
suggestion: Could you add a comment, how versions are sorted/which properties the sorted result has?

One can understand the code of the sorting. It is a kind of topological sorting.
But still, it is hard to understand, which properties of the sorting are essential and guaranteed and which are only accidental.
One needs to follow and understand the existing usages, and even than that might not contain all the intended usages.